components/match2/wikis/counterstrike/match_summary.lua

```diff
@@ -0,0 +1,499 @@
+---
+-- @Liquipedia
+-- wiki=counterstrike
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local VodLink = require('Module:VodLink')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+local MatchSummary = Lua.import('Module:MatchSummary/Base', {requireDevIfEnabled = true})
+
+local GREEN_CHECK = '[[File:GreenCheck.png|14x14px|link=]]'
+local NO_CHECK = '[[File:NoCheck.png|link=]]'
+local ARROW_LEFT = '[[File:Arrow sans left.svg|15x15px|link=|Left team starts]]'
+local ARROW_RIGHT = '[[File:Arrow sans right.svg|15x15px|link=|Right team starts]]'
+
+local EPOCH_TIME = '1970-01-01 00:00:00'
+local EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
+
+-- Score Class
+local Score = Class.new(
+	function(self)
+		self.root = mw.html.create('div'):css('width','70px'):css('text-align', 'center')
+		self.table = self.root:tag('table'):css('line-height', '29px')
+		self.top = mw.html.create('tr')
+		self.bottom = mw.html.create('tr')
+	end
+)
+
+function Score:setLeft()
+	self.table:css('float', 'left')
+	return self
+end
+
+function Score:setRight()
+	self.table:css('float', 'right')
+	return self
+end
+
+function Score:setMapScore(score)
+	local mapScore = mw.html.create('td')
+	mapScore
+		:attr('rowspan', '2')
+		:css('font-size', '16px')
+		:css('width', '25px')
+		:wikitext(score or '')
+
+	self.top:node(mapScore)
+
+	return self
+end
+
+function Score:setFirstHalfScore(score, color)
+	local halfScore =  mw.html.create('td')
+	halfScore
+		:addClass('brkts-popup-body-match-sidewins')
+		:css('color', color)
+		:wikitext(score)
+
+	self.top:node(halfScore)
+	return self
+end
```

Comment:
hardcoded color!
maybe something like:
```suggestion
local SIDE_T = 't'
local SIDE_CT = 'ct'

function Score:setFirstHalfScore(score, side)
	local halfScore =  mw.html.create('td')
	halfScore
		:addClass('brkts-popup-body-match-sidewins')
		:addClass('brkts-cs-score-color-' .. (side == SIDE_CT and SIDE_CT or SIDE_T))
		:wikitext(score)

	self.top:node(halfScore)
	return self
end
```

Warning: we would need to create those classes!

Reply:
we can just use `side` instead of `(side == SIDE_CT and SIDE_CT or SIDE_T)`

Reply:
css classes added: https://liquipedia.net/commons/index.php?title=MediaWiki%3ACommon.css%2FBrackets.css&type=revision&diff=441187&oldid=440935

Reply:
> we can just use `side` instead of `(side == SIDE_CT and SIDE_CT or SIDE_T)`

`side == SIDE_CT and SIDE_CT or SIDE_T` was the equivalent of what was used in `CustomMatchSummary._getSideColor`
i am not opposed to allowing plain input
just keep in mind that if it is not (lower cased) `'t'` or `'ct'` it will add a class that doesn't exist^^

and you also need to avoid nill in that case,
so `side or ''` if the possibility exists that it is ever nil (e.g. if not entered)

Reply:
In Input Custom its ensured that the only values entered to txsides are 'ct' or 't'

Reply:
but could it be empty?